src/utils/getTokenMetadata.ts

```diff
@@ -1,9 +1,16 @@
-import { RainbowToken } from '@/entities';
+import { TokenMetadata } from '@/entities/tokens';
 import { rainbowTokenList } from '@/references';
 
 export default function getTokenMetadata(
   tokenAddress: string | undefined
-): RainbowToken | undefined {
+): TokenMetadata | undefined {
   if (!tokenAddress) return undefined;
-  return rainbowTokenList.RAINBOW_TOKEN_LIST[tokenAddress.toLowerCase()];
+  const metadata: TokenMetadata =
+    rainbowTokenList.RAINBOW_TOKEN_LIST[tokenAddress.toLowerCase()];
+
+  // delete chain specific metadata
+  delete metadata?.decimals;
+  delete metadata?.chainId;
+
+  return metadata;
```

Comment:
If the type for `TokenMetadata` matches what's in `RAINBOW_TOKEN_LIST`, then that's great. But `getTokenMetadata` should return `Omit<TokenMetadata, 'decimals' | 'chainId'>` or you should have a separate types for each. That way it's clear that we go from one shape to another.